pkg/storage/unified/sql/backend.go

```diff
@@ -631,6 +632,99 @@ func (b *backend) listLatest(ctx context.Context, req *resourcepb.ListRequest, c
 	return iter.listRV, err
 }
 
+// ListModifiedSince will return all resources that have changed since the given resource version.
+// If a resource has changes, only the latest change will be returned.
+func (b *backend) ListModifiedSince(ctx context.Context, key resource.NamespacedResource, sinceRv int64) (int64, iter.Seq2[*resource.ModifiedResource, error]) {
+	tx, err := b.db.BeginTx(ctx, RepeatableRead)
+	if err != nil {
+		return 0, func(yield func(*resource.ModifiedResource, error) bool) {
+			yield(nil, err)
+		}
+	}
+
+	// Fetch latest RV within the transaction
+	latestRv, err := b.fetchLatestRV(ctx, tx, b.dialect, key.Group, key.Resource)
+	if err != nil {
+		terr := tx.Rollback()
+		if terr != nil {
+			b.log.Warn("Error rolling back transaction in ListModifiedSince", "error", terr)
+		}
+		return 0, func(yield func(*resource.ModifiedResource, error) bool) {
+			yield(nil, err)
+		}
+	}
+
+	// since results are sorted by name ASC and rv DESC, we can get away with tracking the last seen
+	lastSeen := ""
+
+	// rollback transaction if iterator not called within 30 seconds
+	rollbackTimer := time.AfterFunc(30*time.Second, func() {
+		if err := tx.Rollback(); err != nil && !errors.Is(err, sql.ErrTxDone) {
+			b.log.Warn("rollback timer error", "err", err)
+		}
+	})
+
+	seq := func(yield func(*resource.ModifiedResource, error) bool) {
+		rollbackTimer.Stop()
+
+		defer func() {
+			// Always rollback the read-only transaction when iterator is done
+			if rollbackErr := tx.Rollback(); rollbackErr != nil {
+				b.log.Warn("Error rolling back transaction in ListModifiedSince", "error", rollbackErr)
+			}
+		}()
+
+		query := sqlResourceListModifiedSinceRequest{
+			SQLTemplate: sqltemplate.New(b.dialect),
+			Namespace:   key.Namespace,
+			Group:       key.Group,
+			Resource:    key.Resource,
+			SinceRv:     sinceRv,
+		}
+
+		rows, err := dbutil.QueryRows(ctx, tx, sqlResourceHistoryListModifiedSince, query)
+		if err != nil {
+			yield(nil, err)
+			return
+		}
+		if rows != nil {
+			defer func() {
+				if cerr := rows.Close(); cerr != nil {
+					b.log.Warn("listSinceModified error closing rows", "error", cerr)
+				}
+			}()
+		}
+
+		for rows.Next() {
+			mr := &resource.ModifiedResource{}
+			if err := rows.Scan(&mr.Key.Namespace, &mr.Key.Group, &mr.Key.Resource, &mr.Key.Name, &mr.ResourceVersion, &mr.Action, &mr.Value); err != nil {
+				if !yield(nil, err) {
+					return
+				}
+				continue
+			}
+
+			// Deduplicate by name
```

Comment:
I would to the comment that resource and group are the same here, if someone is curious why we don't deduplicate by those (like I did for a moment).